### DISCUSSION TITLE
Missing systems

### DISCUSSION THREAD

**Discussion by willoucom** [retronas/discussions#80] [General]
I noticed some systems missing, for example:
- sega game gear
- sega megadrive 32x
- bandai wonderswan color
- snk neogeo pocket color
- nec supergrafx (yes, that one with 5 games)

I will submit a PR in few hours, reply to this tread if you have found more missing systems

**Comment by birdybro** [comment]
Strange, the MiSTer directory at least has WonderSwan.

**Comment by willoucom** [comment]
wonderswan is already present, but the wonderswan color is not the same system as the wonderswan (processor/ram got an upgrade with the color). 
same issue with gameboy and gameboy color, or neogeo pocket and neogeo pocket color.

**Comment by birdybro** [comment]
If you are talking about the MiSTer folder structure, there is no need for a WonderSwan Color directory. Those directories are mapped to the MiSTer core. The WonderSwan core is also a WonderSwan Color core and when you select "Load Rom" it starts in `/games/Wonderswan` by default. Personally, for my own organization, I use subfolders within the core's folders for this. For instance, I would do something like having `/media/games/Wonderswan/WS` and `/media/games/Wonderswan/WSC` to draw this distinction. Otherwise in the mister navigation you would click "Load ROM" and then have to go up a level and scroll down to a separate higher level folder every time you changed WSC games.

The same applies to Game Gear and SuperGrafX. Game Gear runs in the SMS directory because it's part of the SMS core. SuperGrafX runs in the TGFX16 directory because it's part of the TurboGrafx-16 core.

**Comment by willoucom** [comment]
I didn't take the time to specify, but I'm talking about the folders in /roms (those sorted by manufacturer)

**Comment by birdybro** [comment]
Ah, that makes more sense. :)

I was a bit confused at first since I'm probably not looking at the same thing as you.

**Comment by danmons** [comment]
Thank you for listing these. I'll add them all in soon. 

Some I'm not sure on - are the Neo Geo Pocket and Pocket Color distinct from a ROMs point of view?   I'm thinking less about categorisation and more about how ROMs would be accessed in emulators.  If they're split on disk, it makes it difficult to access from a single menu in an emulator.

I have this same challenge with GameBoy and GameBoy Color. There are distinct emulators, however you often want all your .gc ROMs in your GBC folder anyway.  

Any thoughts on that are appreciated. 

[edit] Took the time to read more above, makes sense.  I'll add the directories in anyway.   We'll see what happens as more emulator systems come on board, and what they require from a symlinking point of view.

**Comment by danmons** [comment]
Systems and links added. 

I had a think about PCEngine too.  I've done some different symlinking to better satisfy MiSTer and other emulators that use the one emulator to deal with multiple "addons" or variations.  

The changes mean the tool will potentially error out now, so people will need to rename their "nec" folder, re-run the tools, then move stuff about to where they want it.

**Comment by willoucom** [comment]
For systems with different names from one region to another, there have always been problems.

We could use a terminology that we have often heard: original name - regional name.
For example: famicom-nes, megadrive-genesis, pcengine-turbographx...

This could be simpler/clearer than symbolic links inside the same manufacturer's folder, which I think should be used with great care.

**Comment by unabletoconnect2429** [comment]
3do seems to be another missing system (unless i'm blind)

**Comment by danmons** [comment]
> 3do seems to be another missing system (unless i'm blind)

Thanks.  Added just now as `roms/other/3do`.  

I'll leave this issue open for people to keep adding systems.

**Comment by sairuk** [comment]
Can you move this to a discussion instead and close out the issue?

**Reply by sairuk** [reply]
Now moved to a discussion

**Comment by unabletoconnect2429** [comment]
before closing can we add a folder for the Dick Smith Wizzard, AKA FunVision, AKA Hanimex Rameses, AKA VTech CreatiVision

**Reply by sairuk** [reply]
We've moved this to a discussion now, it will remain open/unlocked for the foreseeable future as a bucket to capture all the system requests and/or corrections

**Comment by bluemistP** [comment]
How about these systems? 
(folder names seem to be common across retropie/batocera/recalbox or any other emulationstation variant)

I'm not familiar with Amstrad but please check if this counts as a separate system or I can place it inside cpc or pcw?
gx4000

These seem to be arcade systems, but not necessarily handled by mame.
atomiswave
fbneo
model2
model3
naomi

**Comment by ghost** [comment]
Just went down a list of folders I use that I didn't see...

Nintendo Switch
Nintendo Wii U
Capcom Play System
Capcom Play System II
Capcom Play System III
Cave
Creatronic Mega Duck
Daphne
Examu Ex-Board
PopCap
Quiz Machines
Sammy Atomiswave
Sega Model 1
Sega Model 2
Sega Model 3
Sega Naomi
Sega SG-1000
Sega ST-V
Sega System 16
Sega System 24
Sega System 32
Sega Triforce
Sega X Board
Sega Y Board
Singe 2
SNK Neo Geo AES
SNK Neo Geo CD
Taito NESiCAxLive
Taito Type X
Tekno Parrot
Tiger Game.com
Windows
Windows Indie
WoW Action Max
ZiNc

**Reply by sairuk** [reply]
Could you please advise how you are using these? what emulator with what systems, what folder structure?

Still need to add MAME softlists, that is going to be a job and a half heh, I've been putting it off.

**Comment by ghost** [comment]
This is how my launchbox is setup, I like the Idea of having all the roms in retronas, even if there isn't a console to load them on, just for organization

1.  Emulators % ls
2. AAE
3. PPSSPP
4. ActionMax	
5. Pinball FX2
6. American Laser Games		
7. RA C64
8. CPS III				
9. RA MSU MD+
10. Cemu				
11. RA PS1
12. Daphne v1.0-2.2.11 (Customized)	
13. RA Spectrum 128K
14. Demul 2019			
15. Redream
16. Demul Arcade			
17. RetroArch
18. Dolphin Triforce		
19. Retroarch Amiga
20. Dolphin Wii			
21. Ryujinx
22. Dolphin-x64			
23. Sega Model 2 Emulator v1.1a
24. EmuCR-Supermodel-r787-x64	
25. Singe 2
26. FS UAE				
27. Super Model R756
28. MAME (Platform Arcade)		
29. TeknoParrot
30. MAME 0.218			
31. WinUAE
32. MAME 0.220			
33. XEMU
34. MAME Software			
35. YUZU EA
36. MAME Zinc	
37. MegaDuck
38. Mesen				
39. PCSX2 1.7.0

**Reply by sairuk** [reply]
Thanks for the example, the more examples we have the more we can understand use cases.

**Comment by ghost** [comment]
also, just my thoughts, maybe no symlinks in the roms folder itself... so if we need to access files direct its easy to sort out. Like nes folder. it took me awhile to figure out what was going on there hehe

edit: I NFS mount my nas to a mini pc running retronas

**Reply by bluemistP** [reply]
> 

Agreed on this. 
retronas is essentially making its own folder structure standard and it would be nice to have it raw. 

Also how about an easy way to manage symlinks? I'm not a technical user but I would like the option to make my own folder structure (some folders are not real systems but emulator-focused, prboom or tyrquake for example)

**Reply by danmons** [reply]
> also, just my thoughts, maybe no symlinks in the roms folder itself... so if we need to access files direct its easy to sort out. Like nes folder. it took me awhile to figure out what was going on there hehe


> retronas is essentially making its own folder structure standard and it would be nice to have it raw.


The current logic is that the folder is always the name of the console in its release territory, and the symlink is the name of regional releases to make it convenient for users who don't know the difference.

"megadrive" and "famicom" tend to confuse a lot of users, for example.  And I don't feel right making them "genesis" and "nes" when they weren't the original system names.

If the answer is to have separate "nes" and "famicom" dirs, that becomes cumbersome to manage for users, as well as most automatic ROM sorting tools.

I want to strike a balance of simplicity and sensibility with these.  The vast majority of users aren't going to want to bother with the nicheness of per-region folders nor self managing symlinks.  Advanced users can get in and do what they want with standard Linux tools.  The goal is to keep it as simple as possible for non-Linux users who just want a functioning NAS with as many per-device shares set up to work as automatically as possible (which means multiple regions inside a single folder to be compatible with what emulators expect).

**Comment by 40wattrange** [comment]
Hello I'd like to request

Philips CDi
3DO (i see above it says it is in /roms/other - but i still do not see it after updating)
Atari Jaguar CD
Nintendo WiiWare (not sure if this folder would actually need to be anything different from the Wii folder itself, or perhaps a sub folder).
